src/slice.cpp

```diff
@@ -2,6 +2,33 @@
 #include "log.hpp"
 #include "power_policy/_power_policy.hpp"
 #include <lib/assert.hpp>
+#include <fstream>
+
+time_point Slice::start_time;
+
+static class ScheduleLogger
+{
+public:
+    ScheduleLogger()
+    {
+        const char *log = getenv("DEMOS_SCHEDULE_LOG");
+        if (!log) return;
+        if constexpr (enabled) {
+            schedule_out.open(log, std::ios_base::trunc);
+        } else {
+            throw runtime_error(
+              "DEMOS_SCHEDULE_LOG env. variable is not supported in release build");
+        }
+    }
+    void log(const std::string &msg)
```

Comment:
I'd prefer to change this to something like `log_process_start`, and pass the relevant parameters, instead of a constructing and passing a string.